team_project_access.go

```diff
@@ -0,0 +1,218 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ TeamProjectAccesses = (*teamProjectAccesses)(nil)
+
+// TeamProjectAccesses describes all the team project access related methods that the Terraform
+// Enterprise API supports
+//
+// TFE API docs: (TODO: ADD DOCS URL)
+// **Note: This functionality is still in BETA and subject to change.**
+type TeamProjectAccesses interface {
+	// List all project accesses for a given project.
+	List(ctx context.Context, options *TeamProjectAccessListOptions) (*TeamProjectAccessList, error)
+
+	// Add team access for a project.
+	Add(ctx context.Context, options TeamProjectAccessAddOptions) (*TeamProjectAccess, error)
+
+	// Read team access by project ID.
+	Read(ctx context.Context, teamProjectAccessID string) (*TeamProjectAccess, error)
+
+	// Update team access on a project.
+	Update(ctx context.Context, teamProjectAccessID string, options TeamProjectAccessUpdateOptions) (*TeamProjectAccess, error)
+
+	// Remove team access from a project.
+	Remove(ctx context.Context, teamProjectAccessID string) error
+}
+
+// teamProjectAccesses implements TeamProjectAccesses
+type teamProjectAccesses struct {
+	client *Client
+}
+
+// TeamProjectAccessType represents a team project access type.
+type TeamProjectAccessType string
+
+const (
+	TeamProjectAccessAdmin TeamProjectAccessType = "admin"
+	TeamProjectAccessRead  TeamProjectAccessType = "read"
+)
+
+// TeamProjectAccessList represents a list of team project accesses
+type TeamProjectAccessList struct {
+	*Pagination
+	Items []*TeamProjectAccess
+}
+
+// TeamProjectAccess represents a project access for a team
+type TeamProjectAccess struct {
+	ID     string                `jsonapi:"primary,team-projects"`
+	Access TeamProjectAccessType `jsonapi:"attr,access"`
+
+	// Relations
+	Team    *Team    `jsonapi:"relation,team"`
+	Project *Project `jsonapi:"relation,project"`
+}
+
+// TeamProjectAccessListOptions represents the options for listing team project accesses
+type TeamProjectAccessListOptions struct {
+	ListOptions
+	ProjectID string `url:"filter[project][id]"`
+}
+
+// TeamProjectAccessAddOptions represents the options for adding team access for a project
+type TeamProjectAccessAddOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,team-projects"`
+	// The type of access to grant.
+	Access *TeamProjectAccessType `jsonapi:"attr,access"`
+
+	// The team to add to the project
+	Team *Team `jsonapi:"relation,team"`
+	// The project to which the team is to be added.
+	Project *Project `jsonapi:"relation,project"`
```

Comment:
If these fields are required, can you make them struct types instead of pointer? (We don't specify omitempty, and I doubt we want to send empty attribute keys)

Reply:
Sure, thanks